### PR TITLE
fix (operator): audits on getBatchFromS3

### LINKED ISSUE #768: fix (operator): audits on get batch from s3
- Fix oom attack with tars
- Fix bypass length

### PR DESCRIPTION
**Pull request by taturosati**
- Fix oom error with gzip bomb
- Fix bypass max batch size

Both were fixed with io.LimitReader

# To test

Follow audit instructions for attack cases, check everything still works as normal

### CODE REVIEW

**Review by NicolasRampoldi — APPROVED** [review]

File: operator/pkg/s3.go [hunk 1]
```diff
@@ -37,11 +38,20 @@ func (o *Operator) getBatchFromS3(batchURL string, expectedMerkleRoot [32]byte)
...
+	// This is to prevent the operator from downloading a larger than expected file
+	// + 1 is added to the contentLength to check if the response body is larger than expected
+	reader := io.LimitedReader{R: resp.Body, N: contentLength + 1}
+	batchBytes, err := io.ReadAll(&reader)
 	if err != nil {
 		return nil, err
 	}
 
+	// Check if the response body is larger than expected
+	if reader.N == 0 {
```
Comment (entropidelic) [inline operator/pkg/s3.go]: in the [io.LimitedReader docs](https://pkg.go.dev/io#LimitedReader) I found
> Read returns EOF when N <= 0 or when the underlying R returns EOF. 

should the condition be changed to 
```suggestion
	if reader.N <= 0 {
```
?

**Comment by entropidelic** [comment]
aside from one comment regarding the `LimitedReader` documentation on its parameter N, code looks good and the attacks were handled succesfully with this solution

File: operator/pkg/s3.go (hunk 1 above)
Comment (MauroToscano) [reply operator/pkg/s3.go]: Seems like it should be changed, yes

**Comment by MauroToscano** [comment]
Is there a zip bomb to test ? We could have it around to try it

**Review by entropidelic — APPROVED** [review]

**Pull request merged**